fast_matched_filter/fast_matched_filter.py

```diff
@@ -92,29 +92,60 @@ def matched_filter(templates, moveouts, weights, data, step, arch='cpu'):
         print("Compiled library for {} not loaded; exiting!".format(arch))
         return
 
-    if templates.ndim == 2:
-        n_templates = np.int32(1)
-    elif templates.ndim == 3 and data.ndim == 3:
+    # figure out and check input formats
+    impossible_dimensions = False
+    if templates.ndim > data.ndim:
+        n_templates = np.int32(templates.shape[0])
+
+        assert templates.shape[1] == data.shape[0] # check stations
+        n_stations = np.int32(templates.shape[1])
+
+        if templates.ndim == 4:
+            assert templates.shape[2] == data.shape[1] # check components
+            n_components = np.int32(templates.shape[2])
+        elif templates.ndim == 3:
+            n_components = np.int32(1)
+        else:
+            impossible_dimensions = True
+
+    elif: templates.ndim == date.ndim:
```

Comment:
typo - date rather than data.
```suggestion
    elif: templates.ndim == data.ndim:
```

Reply:
Fixed!